src/main/kotlin/uk/gov/dluhc/printapi/messaging/service/PrintResponseProcessingService.kt

```diff
@@ -115,37 +110,13 @@ class PrintResponseProcessingService(
 
         certificateRepository.save(certificate)
 
-        sendEmailIfCertificateNotDelivered(newStatus, certificate, printResponse.requestId)
+        sendOnlyFirstMatchingEmail(
+            { certificateNotDeliveredEmailSender.testAndSend(printResponse, certificate) },
```

Comment:
I think it's better to check the print status everytime before invoking both these calls here and within those methods, you could still safeguard it if needed